### DISCUSSION TITLE
Terragrunt with TFE

### DISCUSSION THREAD

**Discussion by geek876** [gruntwork-io/knowledge-base#829] [How do I]
Team,
Trying to get TG work with TFE, especially for dependencies.
I am aware that with TFE, the standard `input` way of terragrunt does'nt work so have to use `generate` block to generate `auto.tfvars` file.

Here is my structure
```
├── alb
│   ├── _dependencies.hcl
│   └── terragrunt.hcl
└── vpc
    └── terragrunt.hcl
```
Module `alb` depends on `vpc` for its `vpc.id`

The `alb/_dependencies.hcl` is 
```

dependencies {
  paths = ["../vpc"]
}


dependency "vpc" {
  config_path = "../vpc"
  mock_outputs = {
    vpc_id  = "mock-vpc-id"
  }
}


inputs = {
  _vpc_id         = dependency.vpc.outputs.vpc_id
}

```
Then to use these `var`, i have `alb/terragrunt.hcl` 
```
locals {
  dependencies = read_terragrunt_config("./_dependencies.hcl")
  alb = {
    vpc_id  = local.dependencies.inputs._vpc_id
  }

  tags = {

  }
}

generate "tfvars-vpc" {
  path      = "vpc.auto.tfvars"
  if_exists = "overwrite"
  disable_signature = true
  contents = <<EOF
alb = ${jsonencode(local.alb)}
EOF
}
```

 
in the TF code I reference `alb.vpc_id`, when i want to use `vpc_id`.

When i run `terragrunt run-all apply` from the root of the folder , it applies `vpc` and then `alb` but the `vpc_id` value is always coming as `mock_vpc_id` and not the actual `vpc_id` output. 

If i first run `terragrunt apply ` from VPC, and then run `terragrunt apply` from alb separaterly, it works.

The backend configuration is as below:
For VPC and ALB, it will be different workspaces.
```
terraform {
    backend "remote" {
        hostname = "${local.tfc_hostname}"
        organization = "${local.tfc_organization}"
        workspaces {
            name = "WS-${local.env}-${local.project}-${local.module}"
        }
    }
}
EOF
}
```

**Comment by denis256** [comment]
Hi,
was tried to set on dependency:
```
mock_outputs_allowed_terraform_commands = ["validate"]
```

https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#dependency
